### PR TITLE
allow key thumbprints as authorized-by

### LINKED ISSUE #51: allow key thumbprints as `authorized-by`
We currently allow raw keys, certified keys and cert chains as `authorized-by` types.

Another form of `authorized-by` could be a key thumbprint (e.g., PSA signer ID).

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #51

### CODE REVIEW

**Comment by nedmsmith** [comment]
Should this PR be updated to use COSE_CertHash = [ hashAlg: (int / tstr), hashValue: bstr ] ?

**Comment by thomas-fossati** [comment]
> Should this PR be updated to use COSE_CertHash = [ hashAlg: (int / tstr), hashValue: bstr ] ?

I'll do the relaxation of the digests in a separate PR.

**Comment by thomas-fossati** [comment]
> > Should this PR be updated to use COSE_CertHash = [ hashAlg: (int / tstr), hashValue: bstr ] ?
> 
> I'll do the relaxation of the digests in a separate PR.

See https://github.com/ietf-rats-wg/draft-ietf-rats-corim/pull/56

File: cddl/cbor-tags.txt [hunk 1]
```diff
@@ -11,4 +11,5 @@ tagged-min-svn = #6.553(min-svn)
 tagged-pkix-base64-key-type = #6.554(tstr)
 tagged-pkix-base64-cert-type = #6.555(tstr)
 tagged-pkix-base64-cert-path-type = #6.556(tstr)
+tagged-thumbprint-type = #6.557(hash-entry)
```
Comment (thomas-fossati) [inline cddl/cbor-tags.txt:14]:
```suggestion
tagged-thumbprint-type = #6.557(digest)
```

File: cddl/crypto-key-type-choice.cddl [hunk 2]
```diff
@@ -1,7 +1,9 @@
 $crypto-key-type-choice /= tagged-pkix-base64-key-type
 $crypto-key-type-choice /= tagged-pkix-base64-cert-type
 $crypto-key-type-choice /= tagged-pkix-base64-cert-path-type
+$crypto-key-type-choice /= tagged-thumbprint-type
 
 tagged-pkix-base64-key-type = #6.554(tstr)
 tagged-pkix-base64-cert-type = #6.555(tstr)
 tagged-pkix-base64-cert-path-type = #6.556(tstr)
+tagged-thumbprint-type = #6.557(hash-entry)
```
Comment (thomas-fossati) [inline cddl/crypto-key-type-choice.cddl:9]:
```suggestion
tagged-thumbprint-type = #6.557(digest)
```

**Comment by thomas-fossati** [comment]
@nedmsmith @yogeshbdeshpande @henkbirkholz should I merge this?

**Review by nedmsmith — APPROVED** [review]

**Pull request merged**